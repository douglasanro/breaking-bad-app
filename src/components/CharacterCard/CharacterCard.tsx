import React from 'react';
import { makeStyles, createStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import ICharacter from 'models/ICharacter';
import getCurrentAgeByDate from 'utils/getCurrentAgeByDate';

type TCharacterCardProps = Pick<ICharacter, 'name' | 'birthday' | 'img'>;

const useStyles = makeStyles(() =>
  createStyles({
    card: {
      maxWidth: 600,
    },
    media: {
      height: 450,
      objectPosition: 'top',
    },
  })
);

const CharacterCard: React.FC<TCharacterCardProps> = ({ name, birthday, img }) => {
  const classes = useStyles();
  const age = getCurrentAgeByDate(birthday);

  return (
    <Card className={classes.card}>
      <CardMedia component="img" image={img} title={name} alt={name} className={classes.media} />
      <CardContent>
        <Typography gutterBottom variant="h5" component="h2">
          {name}
        </Typography>
        <Typography variant="body2" color="textSecondary" component="p">
          Age: {age}
        </Typography>
      </CardContent>
    </Card>
  );
};

export default React.memo(CharacterCard);
