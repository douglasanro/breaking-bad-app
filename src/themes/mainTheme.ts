import { unstable_createMuiStrictModeTheme as createMuiTheme } from '@material-ui/core';

const mainTheme = createMuiTheme({
  palette: {
    primary: {
      light: '#69A704',
      main: '#3A7116',
      dark: '#064500',
      contrastText: '#FFF',
    },
    secondary: {
      light: '#FFFF51',
      main: '#DFD400',
      dark: '#A9A300',
      contrastText: '#000',
    },
  },
  overrides: {
    MuiCssBaseline: {
      "@global": {
        "*, *::before, *::after": {
          boxSizing: "content-box",
        },
        body: {
          background: '#FFF'
        },
      },
    },
  },
});

export default mainTheme;
