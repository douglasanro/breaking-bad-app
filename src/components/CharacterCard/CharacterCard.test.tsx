import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/core/styles';
import CharacterCard from './CharacterCard';
import mainTheme from 'themes/mainTheme';
import { mockCharacter } from 'setupTests';

const setupRender = ({ name, birthday, img }: React.ComponentProps<typeof CharacterCard>) =>
  render(<CharacterCard name={name} birthday={birthday} img={img} />, {
    wrapper: ({ children }) => <ThemeProvider theme={mainTheme}>{children}</ThemeProvider>,
  });

test('Render CharacterCard correctly', () => {
  const { container } = setupRender({
    name: mockCharacter.name,
    birthday: mockCharacter.birthday,
    img: mockCharacter.img,
  });
  expect(container).toMatchSnapshot();
});
