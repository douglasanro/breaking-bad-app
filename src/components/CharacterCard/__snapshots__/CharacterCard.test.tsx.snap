// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render CharacterCard correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <img
      alt="Walter White"
      class="MuiCardMedia-root makeStyles-media-2 MuiCardMedia-media MuiCardMedia-img"
      src="https://images.amcnetworks.com/amc.com/wp-content/uploads/2015/04/cast_bb_700x1000_walter-white-lg.jpg"
      title="Walter White"
    />
    <div
      class="MuiCardContent-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Walter White
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Age: 
        63 years
      </p>
    </div>
  </div>
</div>
`;
